assembly/src/v1/parsers/io_ops.rs

```diff
@@ -50,7 +50,218 @@ pub fn parse_readw(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), As
 // RANDOM ACCESS MEMORY
 // ================================================================================================
 
-/// TODO: implement
-pub fn parse_mem(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), AssemblyError> {
-    unimplemented!()
+/// Appends STOREW or LOADW and required stack manipulations to the span block as specified by the memory
+/// operation. If the op does not contain an address, the memory address is assumed to be on top of the stack.
+/// Otherwise, the provided address must be pushed so it is on top of the stack when STOREW or LOADW is executed
+///
+/// push: reads a word (4 elements) from memory and pushes it onto the stack. If the address was already
+/// on the stack, POP it off first. Then PAD 4 times to make space for the new elements, MOVUP5 the memory
+/// address to the top of the stack, and LOADW to load the word from memory into the padded spots
+///
+/// load: reads a word from memory and overwrites the top 4 elements of the stack. If an address was
+/// provided, PUSH the address, then LOADW to load a word from memory.
+///
+/// pop: pops the top 4 elements off the stack and saves them to memory. If an address was provided,
+/// PUSH the address first, STOREW to store the top 4 elements (below the address) at the specified memory address,
+/// then DROP 4 times to pop the elements off the stack
+///
+/// store: saves the top 4 elements of the stack to memory and leaves them on the stack. If an address was
+/// provided, PUSH the address first, then STOREW to save the top 4 elements to the specified memory location.
+pub fn parse_mem(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
```

Comment:
nit: I usually try to keep lines width within 100 chars. also, some sentences seem to be missing periods.

Reply:
Fixed.